platform/ios/detect.py

```diff
@@ -146,12 +141,10 @@ def configure(env):
         env.Append(ASFLAGS=["-arch", "arm64"])
         env.Append(CPPDEFINES=["NEED_LONG_INT"])
 
-    # Disable exceptions on non-tools (template) builds
-    if not env["tools"]:
```

Comment:
I removed the check since we don't support editor build on iOS for now. Could be readded once it's needed.